WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeWebActivity.java

```diff
@@ -0,0 +1,148 @@
+package org.wordpress.android.ui.themes;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.Toast;
+
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.models.Blog;
+import org.wordpress.android.models.Theme;
+import org.wordpress.android.ui.WPWebViewActivity;
+import org.wordpress.android.util.AppLog;
+
+public class ThemeWebActivity extends WPWebViewActivity {
+    public static final String IS_CURRENT_THEME = "is_current_theme";
+    public static final String IS_PREMIUM_THEME = "is_premium_theme";
+    public static final String THEME_NAME = "theme_name";
+    private static final String THEME_DOMAIN_PUBLIC = "pub";
+    private static final String THEME_DOMAIN_PREMIUM = "premium";
+    private static final String THEME_URL_PREVIEW = "%s/wp-admin/customize.php?theme=%s/%s&hide_close=true";
+    private static final String THEME_URL_SUPPORT = "https://wordpress.com/themes/%s/support/?preview=true&iframe=true";
+    private static final String THEME_URL_DETAILS = "https://wordpress.com/themes/%s/%s/?preview=true&iframe=true";
+    private static final String THEME_URL_DEMO_PARAMETER = "demo=true&iframe=true&theme_preview=true";
+    private static final String THEME_HTTPS_PREFIX = "https://";
+
+    public enum ThemeWebActivityType {
+        PREVIEW,
+        DEMO,
+        DETAILS,
+        SUPPORT
+    }
+
+    public static void openTheme(Activity activity, String themeId, ThemeWebActivityType type, boolean isCurrentTheme) {
+        String blogId = WordPress.getCurrentBlog().getDotComBlogId();
+        Theme currentTheme = WordPress.wpDB.getTheme(blogId, themeId);
+        String url = getUrl(activity, currentTheme, type, currentTheme.isPremium());
+
+        openWPCOMURL(activity, url, currentTheme, WordPress.getCurrentBlog(), isCurrentTheme);
+    }
+
+    private static void openWPCOMURL(Activity activity, String url, Theme currentTheme, Blog blog, Boolean isCurrentTheme) {
+        if (activity == null) {
+            AppLog.e(AppLog.T.UTILS, "Context is null");
+            return;
+        }
+
+        if (TextUtils.isEmpty(url)) {
+            AppLog.e(AppLog.T.UTILS, "Empty or null URL passed to openUrlByUsingMainWPCOMCredentials");
```

Comment:
I remember I used that long name to be sure we will not use it to open an external URL (as it happened in the past). (and wpcom token sent to 3rd party server).